src/main/java/Phoebe.java

```diff
@@ -0,0 +1,199 @@
+        } catch (FileNotFoundException e) {
+            System.out.println("No saved tasks found. Starting fresh.");
+        } catch (Exception e) {
+            System.out.println("Error loading tasks. Data file might be corrupted.");
+        }
+    }
+    private static Task parseTaskFromString(String data) {
+        String[] parts = data.split("\\|");
+        // Basic validation
+        if (parts.length < 3) return null;
+
+        String type = parts[0].trim();
+        boolean isDone = parts[1].trim().equals("1");
+        String description = parts[2].trim();
+        Task task = null;
+
+        switch (type) {
+            case "T":
+                task = new ToDo(description);
+                break;
+            case "D":
+                if (parts.length < 4) return null;
+                String by = parts[3].trim();
+                task = new Deadline(description, by);
+                break;
+            case "E":
+                if (parts.length < 5) return null;
+                String times = parts[3].trim(); // Assuming "from to to" format
+                String[] timeParts = times.split(" to ", 2);
+                if (timeParts.length < 2) return null; // Proper validation
+                task = new Event(description, timeParts[0], timeParts[1]);
+                break;
+        }
+
+        if (task != null && isDone) {
+            task.markAsDone();
+        }
+        return task;
+    }
+
+
+
+
+    private static void addTask(String input) throws PhoebeException {
+        input = input.trim();
+        Task newTask = null;
+        if (input.toLowerCase().startsWith("todo")) {
+            String description = input.substring(4).trim();
+            if (description.isEmpty()) {
+                throw new PhoebeException("hello fool how can you todo nothing??????");
+            }
+            newTask = new ToDo(description);
+        } else if (input.toLowerCase().startsWith("deadline")) {
+            String[] parts = input.substring(8).split("/by", 2);
+            if (parts.length < 2 || parts[0].trim().isEmpty()) {
+                throw new PhoebeException("if never tell me /by how I know the deadline??? im ded.");
+            }
+            newTask = new Deadline(parts[0].trim(), parts[1].trim());
+        } else if (input.toLowerCase().startsWith("event")) {
+            String[] parts = input.substring(5).split("/from", 2);
+            if (parts.length < 2 || parts[0].trim().isEmpty()) {
+                throw new PhoebeException("you tell me empty event for what");
+            }
+            String[] timeParts = parts[1].trim().split("/to", 2);
+            if (timeParts.length < 2) {
+                throw new PhoebeException("event no (/from and /to)??? then what time u can go home?");
+            }
+            newTask = new Event(parts[0].trim(), timeParts[0].trim(), timeParts[1].trim());
+        } else {
+            throw new PhoebeException("you dont make any sense");
+        }
```

Comment:
Consider eliminating the arrow head code here. It requires quite some bit of refactoring,